crates/gen/src/types/class.rs

```diff
@@ -121,6 +129,17 @@ impl Class {
             let async_get = async_get_tokens(&self.name, &self.interfaces);
             let debug = debug::debug_tokens(&self.name, &self.interfaces);
 
+            let send_sync = if self.is_agile {
+                let constraints = &self.name.constraints;
+                let name = &self.name.tokens;
+                quote! {
+                    unsafe impl<#constraints> ::std::marker::Send for #name {}
+                    unsafe impl<#constraints> ::std::marker::Sync for #name {}
```

Comment:
I'm convinced that deriving Send is find, but I'm wondering how sure we can be about Sync. @kennykerr the semantics on `Sync` are not 100% fully documented but in short a type is `Sync` if `&T` can be safely sent between threads and there are never any unsynchronized reads and writes to those references.

Reply:
If we're unsure, it would probably still be useful to just implement `Send` while figuring out `Sync`, but I don't know enough about the semantics of the underlying types to have much input beyond that.

Reply:
Yes, any type that is agile should be both `Send` and `Sync`. WinRT types should be implemented in a manner that is thread safe both in terms of their COM lifetime (thread-safe reference counting) and their internal state (thread-safe mutation). There are types that may violate this definition, but we should regard those as bugs.